app/models/ext_management_system.rb

```diff
@@ -503,4 +503,23 @@ def blacklisted_event_names
   def self.blacklisted_events
     BlacklistedEvent.where(:provider_model => name, :ems_id => nil)
   end
+
+  def self.credential_validation(ems, auth_type, creds, revalidate)
```

Comment:
Since you are passing in the ems, then this doesn't need to be a class method.  You can make it an instance method, remove the ems parameter, and just call it on the ems from the caller: `ems.credential_validation(auth_type, creds, revalidate)`.  The task method can be adjusted similarly.  The queue item can be instance based as well, by putting `:instance_id => ems.id` in the queue_opts.